nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DetectDuplicateRecord.java

```diff
@@ -0,0 +1,620 @@
+            .allowableValues(
+                    MD5_ALGORITHM_VALUE,
+                    SHA1_ALGORITHM_VALUE,
+                    SHA256_ALGORITHM_VALUE,
+                    SHA512_ALGORITHM_VALUE
+            )
+            .defaultValue(SHA1_ALGORITHM_VALUE.getValue())
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor FILTER_TYPE = new PropertyDescriptor.Builder()
+            .name("Filter Type")
+            .description("The filter used to determine whether a record has been seen before based on the matching RecordPath criteria.")
+            .allowableValues(
+                    HASH_SET_VALUE,
+                    BLOOM_FILTER_VALUE
+            )
+            .defaultValue(HASH_SET_VALUE.getValue())
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor FILTER_CAPACITY_HINT = new PropertyDescriptor.Builder()
+            .name("Filter Capacity Hint")
+            .description("An estimation of the total number of unique records to be processed. " +
+                    "The more accurate this number is leads to less resizing operations on a HashSet filter and fewer false negatives on a BloomFilter. " +
+                    "Using the CountText processor block first and then using the ${text.line.count} attribute may be ideal for more dynamic estimates.")
+            .defaultValue("25000")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor BLOOM_FILTER_FPP = new PropertyDescriptor.Builder()
+            .name("BloomFilter Probability")
+            .description("The desired false positive probability when using the BloomFilter filter type. " +
+                    "Using a value of .05 for example, means we can be 95% certain that the element is in the filter, " +
+                    "and there is a five-percent probability that the result is a false positive. " +
+                    "When the filter returns false, we can be 100% certain that the element is not present.")
+            .defaultValue("0.10")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .addValidator(StandardValidators.NUMBER_VALIDATOR)
+            .required(false)
+            .build();
+
+
+    // RELATIONSHIPS
+
+    static final Relationship REL_DUPLICATE = new Relationship.Builder()
+            .name("duplicate")
+            .description("Records detected as duplicates in the FlowFile content will be routed to this relationship")
+            .build();
+
+    static final Relationship REL_NON_DUPLICATE = new Relationship.Builder()
+            .name("non-duplicate")
+            .description("If the record was not found in the cache, it will be routed to this relationship")
```

Comment:
might want to remove 'cache' from this description and instead just indicate 'If the record was not detected as a duplicate it will be routed to this relationship'

Reply:
This wording was copied from the `DetectDuplicate` processor. It still sounds accurate because we are checking against the cache.